app/models/mixins/event_mixin.rb

```diff
@@ -63,7 +63,7 @@ def find_one_event(assoc, order)
 
   module ClassMethods
     def ems_event_filter_column
-      @ems_event_filter_column ||= reflect_on_association(:ems_events).try(:foreign_key) || name.foreign_key
+      @ems_event_filter_column ||= reflect_on_association(:ems_events).try(:foreign_key) || base_model.name.foreign_key
```

Comment:
Is there a reason why the `ems_events` reflection isn't present for all base classes?

Reply:
see the other discussion, they should be fixed but not here.